%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% This is just an example/guide for you to refer to when submitting manuscripts to Frontiers, it is not mandatory to use Frontiers .cls files nor frontiers.tex  %
% This will only generate the Manuscript, the final article will be typeset by Frontiers after acceptance.
%                                              %
%                                                                                                                                                         %
% When submitting your files, remember to upload this *tex file, the pdf generated with it, the *bib file (if bibliography is not within the *tex) and all the figures.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% Version 3.4 Generated 2018/06/15 %%%
%%% You will need to have the following packages installed: datetime, fmtcount, etoolbox, fcprefix, which are normally inlcuded in WinEdt. %%%
%%% In http://www.ctan.org/ you can find the packages and how to install them, if necessary. %%%

\documentclass[utf8]{frontiersSCNS}

%\setcitestyle{square} % for Physics and Applied Mathematics and Statistics articles
\usepackage{url,hyperref,lineno,microtype,subcaption}
\usepackage[onehalfspacing]{setspace}

\linenumbers


% BELOW TAKEN FROM rticles plos template
%
% amsmath package, useful for mathematical formulas
\usepackage{amsmath}
% amssymb package, useful for mathematical symbols
\usepackage{amssymb}

% hyperref package, useful for hyperlinks
\usepackage{hyperref}

% graphicx package, useful for including eps and pdf graphics
% include graphics with the command \includegraphics
\usepackage{graphicx}

% Sweave(-like)
\usepackage{fancyvrb}
\DefineVerbatimEnvironment{Sinput}{Verbatim}{fontshape=sl}
\DefineVerbatimEnvironment{Soutput}{Verbatim}{}
\DefineVerbatimEnvironment{Scode}{Verbatim}{fontshape=sl}
\newenvironment{Schunk}{}{}
\DefineVerbatimEnvironment{Code}{Verbatim}{}
\DefineVerbatimEnvironment{CodeInput}{Verbatim}{fontshape=sl}
\DefineVerbatimEnvironment{CodeOutput}{Verbatim}{}
\newenvironment{CodeChunk}{}{}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

\usepackage{color}

\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

% Below is from frontiers
%
\bibliographystyle{frontiersinSCNS}
% Use doublespacing - comment out for single spacing
%\usepackage{setspace}
%\doublespacing


% Leave a blank line between paragraphs instead of using \\


\def\keyFont{\fontsize{8}{11}\helveticabold }


%% ** EDIT HERE **
%% PLEASE INCLUDE ALL MACROS BELOW

%% END MACROS SECTION

%\usepackage[table]{xcolor}
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}


\def\Authors{
  Dennis Psaroudakis\,\textsuperscript{1},
  Ha Vu\,\textsuperscript{1},
  Colleen Yanarella\,\textsuperscript{1},
  Steven Cannon\,\textsuperscript{1},
  Darwin Campbell\,\textsuperscript{1},
  Parnal Joshi\,\textsuperscript{1},
  Iddo Friedberg\,\textsuperscript{1,4},
  Kokulapalan Wimalanathan\,\textsuperscript{1,2},
  Carolyn J. Lawrence-Dill\,\textsuperscript{1,2,3*}}

\def\Address{

  \textsuperscript{1} Bioinformatics and Computational Biology, Iowa State University,  Ames,  IA,  USA
  
  \textsuperscript{2} Department of Genetics, Development, and Cell Biology, Iowa State University,  Ames,  IA,  USA
  
  \textsuperscript{3} Department of Agronomy, Iowa State University,  Ames,  IA,  USA
  
  \textsuperscript{4} Department of Veterinary Microbiology, Iowa State University,  Ames,  IA,  USA
  }

  
  \def\firstAuthorLast{Psaroudakis {et~al.}}
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  \def\corrAuthor{Carolyn J. Lawrence-Dill}\def\corrAddress{Roy J. Carver Co-Lab, Iowa State University\\1111 WOI Rd\\Ames, State IA, 50011 USA}\def\corrEmail{\href{mailto:triffid@iastate.edu}{\nolinkurl{triffid@iastate.edu}}}
  


\begin{document}
\onecolumn
\firstpage{1}

\title[Short Title]{Article Title}
\author[\firstAuthorLast]{\Authors}
\address{} %This field will be automatically populated
\correspondance{} %This field will be automatically populated

\extraAuth{}% If there are more than 1 corresponding author, comment this line and uncomment the next one.
%\extraAuth{corresponding Author2 \\ Laboratory X2, Institute X2, Department X2, Organization X2, Street X2, City X2 , State XX2 (only USA, Canada and Australia), Zip Code2, X2 Country X2, email2@uni2.edu}


\maketitle

\begin{abstract}

Abstract length and content varies depending on article type. Refer to 
\url{http://www.frontiersin.org/about/AuthorGuidelines} for abstract requirement
and length according to article type.

%All article types: you may provide up to 8 keywords; at least 5 are mandatory.
\tiny
 \keyFont{ \section{Keywords:} Text Text Text Text Text Text Text Text } 

\end{abstract}

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

Hello, how are we doing?

\hypertarget{methods}{%
\section{Methods}\label{methods}}

\hypertarget{clean-up}{%
\subsection{Clean up}\label{clean-up}}

All functional annotation sets were cleaned up the following way (using definitions from the Gene Ontology version 2019-07-01):

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Any annotations where the GO accession was marked as obsolete were removed.
\item
  Some terms in the GO have `alternative ids'. When naively removing duplicates, two entries will not be recognized as duplicates if they have different accessions pointing to the same GO term. Therefore, all GO accessions were changed to their respecitve `main id' and the dataset was again scanned for duplicates.
\end{enumerate}

Table 1 provides information on the number of annotations that were removed this way from each dataset.
All further analyses were performed on the cleaned datasets since we assume the user will only be interested in still valid and non-redundant functional annotations.

\hypertarget{choosing-the-right-evaluation-metric}{%
\subsection{Choosing the right evaluation metric}\label{choosing-the-right-evaluation-metric}}

A plethora of different metric to evaluate the quality of functional annotation predictions is available using different approaches and there seems to be no clear standard yet. \texttt{TODO:\ THIS\ IS\ WEAK}
Additionally, each of the metrics has a different focus and lalala so choosing a metric for quality evaluation is not trivial.
When we first published GOMAP \citep{Wimalanathan2018}, we used a modified version of the hierarchical evaluation metrics originally introduced in \citep{Verspoor2006} because they were simple, clear, and part of an earlier attempt at unifying and standardizing GO annotation comparisons \citep{Defoin-Platel2011}.
In the meantime, \citet{Plyusnin2018} have published an approach for evaluating different metrics showing substantial differences within the robustness of different approaches.
\texttt{TODO\ DESCRIBE\ THEIR\ APPROACH}
We have applied their method on the Gold Standards available to us to determine which evaluation metric is the most appropriate in our case.
The results of this analysis can be seen in \texttt{TODO}.

We then evaluated our predictions and the other annotation sets using the best performing metrics as well as the one we previously used.
\texttt{TODO}

\hypertarget{results}{%
\section{Results}\label{results}}

\ldots{} a quantitative comparison of the datasets in Table.

\begin{table}[t]

\caption{\label{tab:cleanup-table}Number of removed annotations during cleanup.}
\centering
\begin{tabular}{llrr}
\toprule
Genome & Dataset & Obsolete Annotations & Duplicates\\
\midrule
\rowcolor{gray!6}  \textit{Medicago truncatula} A17 & GOMAP & 0 & 0\\
\cmidrule{1-4}
\textit{Medicago truncatula} R108 & GOMAP & 0 & 0\\
\cmidrule{1-4}
\rowcolor{gray!6}  \textit{Oriza sativa} & GOMAP & 0 & 10\\
\cmidrule{1-4}
 & GOMAP & 111 & 2\\

\rowcolor{gray!6}   & GoldStandard & 38 & 556\\

\multirow{-3}{*}{\raggedright\arraybackslash \textit{Oryza sativa}} & Gramene61-IEA & 10 & 14\\
\cmidrule{1-4}
\rowcolor{gray!6}  \textit{Phaseolus vulgaris} & GOMAP & 0 & 0\\
\cmidrule{1-4}
 & GOMAP & 285 & 0\\

\rowcolor{gray!6}   & GoldStandard & 0 & 10\\

\multirow{-3}{*}{\raggedright\arraybackslash \textit{Triticum aestivum}} & Gramene61-IEA & 47 & 48\\
\cmidrule{1-4}
\rowcolor{gray!6}  \textit{Vigna unguiculata} & GOMAP & 0 & 0\\
\cmidrule{1-4}
 & GOMAP & 1107 & 70\\

\rowcolor{gray!6}   & GoldStandard & 1 & 0\\

 & Gramene49 & 94 & 2\\

\rowcolor{gray!6}  \multirow{-4}{*}{\raggedright\arraybackslash \textit{Zea mays} B73.v3} & Phytozome & 54 & 0\\
\cmidrule{1-4}
 & GOMAP & 752 & 83\\

\rowcolor{gray!6}   & GoldStandard & 55 & 174\\

\multirow{-3}{*}{\raggedright\arraybackslash \textit{Zea mays} B73.v4} & Gramene61-IEA & 99 & 157\\
\cmidrule{1-4}
\rowcolor{gray!6}  \textit{Zea mays} Mo17 & GOMAP & 726 & 77\\
\cmidrule{1-4}
\textit{Zea mays} PH207 & GOMAP & 798 & 76\\
\cmidrule{1-4}
\rowcolor{gray!6}  \textit{Zea mays} W22 & GOMAP & 754 & 82\\
\bottomrule
\end{tabular}
\end{table}

\begin{table}[t]

\caption{\label{tab:annotation-quantities}Quantitative metrics of the cleaned functional annotation sets. CC, BF, MP, and A refer to the aspects of the GO: Cellular Component, Biological Function, Molecular Process, and Any/All.}
\centering
\resizebox{\linewidth}{!}{
\begin{threeparttable}
\begin{tabular}{lrlrrr>{\bfseries}r|rrr>{\bfseries}r|rrr>{\bfseries}r}
\toprule
\multicolumn{3}{c}{ } & \multicolumn{4}{c}{Annotations\textsuperscript{a}} & \multicolumn{4}{c}{Annotated Genes [\%]\textsuperscript{b}} & \multicolumn{4}{c}{Median Ann. per G.\textsuperscript{c}} \\
\cmidrule(l{3pt}r{3pt}){4-7} \cmidrule(l{3pt}r{3pt}){8-11} \cmidrule(l{3pt}r{3pt}){12-15}
Genome & Genes & Dataset & CC & BF & MP & A & CC & BF & MP & A & CC & BF & MP & A\\
\midrule
\rowcolor{gray!6}  \textit{Arachis hypogaea} &  & GOMAP & 153433 & 132944 & 493799 & 780176 & 57667 & 56855 & 67123 & 67124 & 2 & 2 & 6 & 10\\

\textit{Glycine max} &  & GOMAP & 129215 & 113827 & 417555 & 660597 & 46020 & 47034 & 52871 & 52872 & 2 & 2 & 6 & 11\\

\rowcolor{gray!6}  \textit{Hordeum vulgarum} &  & GOMAP & 88130 & 80282 & 272823 & 441235 & 35237 & 36470 & 39733 & 39734 & 2 & 2 & 5 & 10\\

\textit{Medicago truncatula} A17 &  & GOMAP & 107362 & 99719 & 364065 & 571146 & 42325 & 43736 & 50443 & 50444 & 2 & 2 & 6 & 10\\

\rowcolor{gray!6}  \textit{Medicago truncatula} R108 &  & GOMAP & 112343 & 108031 & 382322 & 602696 & 40332 & 50220 & 55706 & 55706 & 1 & 2 & 5 & 9\\

 &  & GOMAP & 72780 & 64685 & 248700 & 386165 & 28619 & 29853 & 35824 & 35825 & 2 & 2 & 6 & 9\\

\rowcolor{gray!6}   &  & GoldStandard & 7730 & 11060 & 19378 & 38176 & 5725 & 7383 & 9031 & 11387 & 1 & 1 & 1 & 3\\

\multirow{-3}{*}{\raggedright\arraybackslash \textit{Oryza sativa}} &  & Gramene61-IEA & 14633 & 32787 & 39105 & 86529 & 10771 & 15537 & 16705 & 21446 & 1 & 1 & 1 & 3\\

\rowcolor{gray!6}  \textit{Phaseolus vulgaris} &  & GOMAP & 72005 & 64583 & 229630 & 366218 & 25934 & 25539 & 27432 & 27433 & 2 & 2 & 6 & 11\\

 &  & GOMAP & 267741 & 218623 & 785960 & 1272324 & 95604 & 98187 & 107890 & 107891 & 2 & 2 & 6 & 10\\

\rowcolor{gray!6}   &  & GoldStandard & 1590 & 923 & 4807 & 7323 & 965 & 620 & 1662 & 1866 & 1 & 0 & 2 & 3\\

\multirow{-3}{*}{\raggedright\arraybackslash \textit{Triticum aestivum}} &  & Gramene61-IEA & 38975 & 109319 & 109518 & 257832 & 28849 & 59596 & 52564 & 75785 & 0 & 1 & 1 & 2\\

\rowcolor{gray!6}  \textit{Vigna unguiculata} &  & GOMAP & 75867 & 68313 & 243278 & 387458 & 27173 & 27124 & 29772 & 29773 & 2 & 2 & 6 & 11\\

 &  & GOMAP & 135211 & 87420 & 291251 & 513882 & 34866 & 38073 & 39468 & 39469 & 3 & 2 & 6 & 11\\

\rowcolor{gray!6}   &  & GoldStandard & 1565 & 65 & 299 & 1929 & 1548 & 60 & 151 & 1634 & 1 & 0 & 0 & 1\\

 &  & Gramene49 & 20072 & 31056 & 30089 & 81217 & 11834 & 17991 & 15800 & 21926 & 1 & 1 & 1 & 3\\

\rowcolor{gray!6}  \multirow{-4}{*}{\raggedright\arraybackslash \textit{Zea mays} B73.v3} &  & Phytozome & 4787 & 19044 & 13100 & 36931 & 4524 & 13728 & 11365 & 16132 & 0 & 1 & 1 & 2\\

 &  & GOMAP & 88827 & 82251 & 278719 & 449797 & 36717 & 37337 & 39323 & 39324 & 2 & 2 & 6 & 10\\

\rowcolor{gray!6}   &  & GoldStandard & 11510 & 15019 & 25737 & 52428 & 8349 & 10067 & 12120 & 14971 & 1 & 1 & 1 & 3\\

\multirow{-3}{*}{\raggedright\arraybackslash \textit{Zea mays} B73.v4} &  & Gramene61-IEA & 20265 & 47657 & 58110 & 126525 & 14774 & 22064 & 23965 & 29152 & 1 & 1 & 2 & 3\\

\rowcolor{gray!6}  \textit{Zea mays} Mo17 &  & GOMAP & 87567 & 79214 & 277787 & 444568 & 33618 & 35105 & 38619 & 38620 & 2 & 2 & 6 & 10\\

\textit{Zea mays} PH207 &  & GOMAP & 90617 & 85500 & 288677 & 464794 & 35170 & 36762 & 40556 & 40557 & 2 & 2 & 6 & 10\\

\rowcolor{gray!6}  \textit{Zea mays} W22 & \multirow{-23}{*}{\raggedleft\arraybackslash 100} & GOMAP & 95390 & 85039 & 289780 & 470209 & 36987 & 37685 & 40689 & 40690 & 2 & 2 & 6 & 10\\
\bottomrule
\end{tabular}
\begin{tablenotes}
\item[a] How many annotations in the CC, BF, and MP aspect does this dataset contain? A = How many in total? $\textrm{A} = \textrm{CC} + \textrm{BF} + \textrm{MP}$
\item[b] How many genes in the genome have at least one GO term from the CC, BF, MP aspect annotated to them? A = How many at least one from any aspect? ($\textrm{A} = \textrm{CC} \cup \textrm{BF} \cup \textrm{MP}$)
\item[c] Take a typical gene that is present in the annotation set. How many annotations does it have in each aspect? A = How many in total? Ask your favorite statistician why $\textrm{A} \neq \textrm{CC} + \textrm{BF} +\textrm{MP}$
\end{tablenotes}
\end{threeparttable}}
\end{table}

\hypertarget{quality-evaluation}{%
\subsection{Quality Evaluation}\label{quality-evaluation}}

\texttt{TODO} If it turns out that our predictions are good with hF but bad with more approriate metrics, explanation would be that score thresholds for the prediction tools used in the GOMAP pipeline have been chosen to maximize this hF value. It now seems reasonable to re-adjust these thresholds to maximize a different metric which will likely result in a drop in hF score but increase in other metrics. Again emphasizes the importance of choosing the right evaluation metric.

\begin{table}[t]

\caption{\label{tab:quality-table}Quality evaluation of the used GO annotation sets.}
\centering
\begin{tabular}{llr}
\toprule
Genome & Dataset & SimGIC2 score\\
\midrule
\rowcolor{gray!6}   & GOMAP & 0.253680\\

\multirow{-2}{*}{\raggedright\arraybackslash \textit{Oryza sativa}} & Gramene61-IEA & 0.330437\\
\cmidrule{1-3}
\rowcolor{gray!6}   & GOMAP & 0.218996\\

\multirow{-2}{*}{\raggedright\arraybackslash \textit{Triticum aestivum}} & Gramene61-IEA & 0.175564\\
\cmidrule{1-3}
\rowcolor{gray!6}   & GOMAP & 0.052182\\

 & Gramene49 & 0.091475\\

\rowcolor{gray!6}  \multirow{-3}{*}{\raggedright\arraybackslash \textit{Zea mays} B73.v3} & Phytozome & 0.028721\\
\cmidrule{1-3}
 & GOMAP & 0.257543\\

\rowcolor{gray!6}  \multirow{-2}{*}{\raggedright\arraybackslash \textit{Zea mays} B73.v4} & Gramene61-IEA & 0.328777\\
\bottomrule
\end{tabular}
\end{table}

\bibliography{GOMAP-Paper-Used}

\end{document}
